serial/uart.c

```diff
@@ -110,6 +169,16 @@ uart_tx_one_char(uint8 uart, uint8 c)
   //Wait until there is room in the FIFO
   while (((READ_PERI_REG(UART_STATUS(uart))>>UART_TXFIFO_CNT_S)&UART_TXFIFO_CNT)>=100) ;
   //Send the character
+  if (UART0 == uart) {
+	if (uart_tx_enable_timer_inited) {
+      // If a tx_completed_interrupt has already been scheduled, cancel it before it fires during our transmission
+	  os_timer_disarm(&uart_tx_enable_timer);
+	} else {
+	  os_timer_setfn(&uart_tx_enable_timer, tx_completed_interrupt, NULL);
+	}
+
+    tx_enable(true);
+  }
```

Comment:
Shouldn't this code be conditional on the tx_enable_pin feature being enabled? It seems to set the timer unconditionally.

Also, I believe this needs to be done with interrupts disabled, else there's a race condition if the interrupt hits between `uart_tx_enable_timer_inited` and  `os_timer_setfn`.